### PR TITLE
Add-to-Playlist Button

### PR DESCRIPTION
**Pull request by narthur**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (mattleff) [inline .husky/pre-commit:1]: Do you need this file?

File: src/components/molecules/favorite.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ export default function Favorite({ id }: { id: string }): JSX.Element {
 	const { isFavorited, toggleFavorited } = useIsFavorited(id);
 
 	return (
-		<button onClick={toggleFavorited}>
+		<button onClick={() => toggleFavorited()}>
```
Comment (mattleff) [inline src/components/molecules/favorite.tsx:9]: Is this a TS issue or why is this needed?

File: src/components/molecules/loadingIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		'1.0': '#ff001e',
+	},
+	barThickness: 7,
+});
+
+const LoadingIndicator = (): JSX.Element | null => {
+	const isFetching = useIsFetching();
+
+	if (isFetching) {
+		console.log('fetching...');
```
Comment (mattleff) [inline src/components/molecules/loadingIndicator.tsx:17]: Nix

File: .husky/pre-commit (hunk 1 above)
Comment (narthur) [reply .husky/pre-commit:1]: Husky added this automatically, so I'm not inclined to delete it...

File: src/components/molecules/playlistButton.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,596 @@
...
+				title: 'playlist2',
+				hasRecording: true,
+			},
+		]);
+
+		// Add second playlist
+
+		await userAddPlaylist('playlist2');
+
+		// Sleep 100ms
```
Comment (mattleff) [inline src/components/molecules/playlistButton.spec.tsx:477]: Comment doesn't match code ;)

File: src/components/molecules/playlistButton.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,596 @@
...
+
+		await sleep({ ms: 200 });
+
+		// Check that first, slow response didn't clobber second, fast response
+
+		expect(getEntry('playlist2')).toBeInTheDocument();
+	});
+
+	it('roles back if mutation fails', async () => {
```
Comment (mattleff) [inline src/components/molecules/playlistButton.spec.tsx:486]:
```suggestion
	it('rolls back if mutation fails', async () => {
```

File: src/components/molecules/playlistButton.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+			lists.map((l: Playlist, i: number) => (
+				<Entry recordingId={recordingId} playlist={l} key={i} />
+			))
+		);
+	};
+
+	return (
+		<>
+			<button data-tip data-for={'playlistButton'}>
+				Add to Playlist
```
Comment (mattleff) [inline src/components/molecules/playlistButton.tsx:66]: i18n

File: src/components/molecules/playlistButton.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+				uuid={'tooltipUuid'}
+				event={'click'}
+				effect={'solid'}
+				clickable={true}
+			>
+				{lists ? (
+					<>
+						<ul className={styles.list}>{getEntries()}</ul>
+						<input
+							placeholder={'New Playlist'}
```
Comment (mattleff) [inline src/components/molecules/playlistButton.tsx:79]: i18n

File: src/components/molecules/playlistButton.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+							value={newPlaylistTitle}
+							onChange={(e) => setNewPlaylistTitle(e.target.value)}
+						/>
+						<label>
+							<input
+								type={'checkbox'}
+								checked={isPublic}
+								onChange={() => setIsPublic(!isPublic)}
+							/>{' '}
+							Public
```
Comment (mattleff) [inline src/components/molecules/playlistButton.tsx:89]: i18n and a few more places below

File: src/containers/sermon/detail.spec.tsx [hunk 9]
```diff
@@ -269,4 +278,34 @@ describe('detailPageGenerator', () => {
...
+
+		const { getByText } = await renderPage();
+
+		expect(getByText('Add to Playlist')).toBeInTheDocument();
+	});
+
+	it('loads fewer sermons in development', async () => {
+		jest.resetModules();
+		const oldEnv = process.env;
+		process.env = {
```
Comment (mattleff) [inline src/containers/sermon/detail.spec.tsx:294]: I would recommend wrapping this test in a `describe()` with a `beforeEach()` and an `afterEach()` that sets and resets `process.env` (like [this](https://stackoverflow.com/a/48042799/168581)). That way if the test fails `process.env` is restored to the prior value.

File: src/lib/api/setPlaylistMembership.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+export async function setPlaylistMembership(
+	recordingId: string,
+	playlistId: string,
+	add: boolean
+): Promise<boolean> {
+	const mutation = add ? mutationAdd : mutationRemove;
+	const variables = { recordingId, playlistId };
+	const result = await fetchApi(mutation, { variables });
+	console.log({ m: 'set membership', recordingId, playlistId, add, result });
```
Comment (mattleff) [inline src/lib/api/setPlaylistMembership.ts:29]: Nix

File: src/lib/api/useIsFavorited.ts [hunk 11]
```diff
@@ -1,39 +1,45 @@
...
+	const { mutate: toggleFavorited } = useMutation(
 		() => {
 			// TODO: Consider using setFavorited response to update data instead of
 			//  invalidating `isFavorited` query
 
-			// TODO: Figure out how to make this data flow less convoluted
-			// onMutate optimistically changes cached data, so we don't
-			// flip faved since it's already been flipped in onMutate
-			return _setFavorited(id, !!isFavorited);
+			return _setFavorited(id, !isFavorited);
```
Comment (mattleff) [inline src/lib/api/useIsFavorited.ts:26]: 👏🏻

File: src/components/molecules/favorite.tsx (hunk 2 above)
Comment (narthur) [reply src/components/molecules/favorite.tsx:9]: The error I get without it:

Screen Shot 2020-12-30 at 4 18 10 PM

**Review by mattleff — APPROVED** [review]
Looks good, just a few comments! 👍🏻

File: src/components/molecules/favorite.tsx (hunk 2 above)
Comment (mattleff) [reply src/components/molecules/favorite.tsx:9]: Ok, that makes sense. The `mutate` function takes arguments which are incompatible with the arguments that `onClick` would give it.
